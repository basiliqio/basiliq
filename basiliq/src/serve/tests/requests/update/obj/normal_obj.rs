use super::*;

crate::run_test_request!(
    obj_single_field,
    Method::PATCH,
    format!("/public__peoples/{}", PEOPLES_IDS[0]),
    200,
    json!({
        "data": json!({
            "type": "public__peoples",
            "id": PEOPLES_IDS[0],
            "attributes": json!({
                "first-name": "OH NO",
            })
        })
    })
);

crate::run_test_request!(
    obj_multiple_fields,
    Method::PATCH,
    format!("/public__peoples/{}", PEOPLES_IDS[0]),
    200,
    json!({
        "data": json!({
            "type": "public__peoples",
            "id": PEOPLES_IDS[0],
            "attributes": json!({
                "first-name": "OH NO",
                "last-name": "OH YES",
                "age": 18,
                "gender": "O"
            })
        })
    })
);

crate::run_test_request!(
    obj_set_to_null,
    Method::PATCH,
    format!("/public__peoples/{}", PEOPLES_IDS[0]),
    200,
    json!({
        "data": json!({
            "type": "public__peoples",
            "id": PEOPLES_IDS[0],
            "attributes": json!({
                "age": Value::Null
            })
        })
    })
);

crate::run_test_request!(
    set_mandatory_to_null,
    Method::PATCH,
    format!("/public__peoples/{}", PEOPLES_IDS[0]),
    400,
    json!({
        "data": json!({
            "type": "public__peoples",
            "id": PEOPLES_IDS[0],
            "attributes": json!({
                "first-name": Value::Null
            })
        })
    })
);

crate::run_test_request!(
    add_relationship_single,
    Method::PATCH,
    format!("/public__peoples/{}", PEOPLES_IDS[0]),
    200,
    json!({
        "data": json!({
            "type": "public__peoples",
            "id": PEOPLES_IDS[0],
            "attributes": json!({
                "age": 100
            }),
            "relationships": json!({
                "public__favorite_color": json!({
                    "data": json!({
                        "id": FAVORITE_COLOR_IDS[2],
                        "type": "public__favorite_color"
                    })
                })
            })
        })
    })
);
