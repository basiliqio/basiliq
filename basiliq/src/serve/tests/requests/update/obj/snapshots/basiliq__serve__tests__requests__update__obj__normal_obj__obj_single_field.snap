---
source: basiliq/src/serve/tests/requests/update/obj/normal_obj.rs
expression: res

---
{
  "jsonapi": {
    "version": "1.0"
  },
  "data": {
    "type": "public__peoples",
    "id": "[uuid]",
    "attributes": {
      "age": 22,
      "first-name": "OH NO",
      "gender": "M",
      "last-name": "Le Roy",
      "twitter": null
    }
  }
}
